google/cloud/aiplatform/metadata/artifact.py

```diff
@@ -0,0 +1,285 @@
+        parent: str,
+        resource: proto.Message,
+        resource_id: str,
+    ) -> proto.Message:
+        return client.create_artifact(
+            parent=parent, artifact=resource, artifact_id=resource_id,
+        )
+
+    @classmethod
+    def update(
+        cls,
+        artifact_id: str,
+        metadata: Optional[Dict] = {},
+        metadata_store_id: Optional[str] = "default",
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ) -> "_Artifact":
+        f"""Updates an Artifact resource.
+
+        Args:
+            artifact_id (str):
+                Required. The {artifact_id} portion of the resource name with
+                the format:
+                projects/{project}/locations/{location}/metadataStores/{metadata_store_id}/artifacts/{artifact_id}.
+            metadata (Dict):
+                Optional. metadata information to update the artifact with.
+            metadata_store_id (str):
+                The {metadata_store_id} portion of the resource name with
+                the format:
+                projects/{project}/locations/{location}/metadataStores/{metadata_store_id}/artifacts/{artifact_id}
+                If not provided, the MetadataStore's ID will be set to "default".
+            project (str):
+                Project where this artifact belongs. Overrides project set in
+                aiplatform.init.
+            location (str):
+                Location where this artifact belongs. Overrides location set in
+                aiplatform.init.
+            credentials (auth_credentials.Credentials):
+                Custom credentials to use to update this artifact. Overrides
+                credentials set in aiplatform.init.
+
+        Returns:
+            artifact (_Artifact):
+                Updated representation of the managed Metadata Artifact resource.
+
+        """
+
+        gapic_artifact = cls(
+            artifact_name=artifact_id,
+            metadata_store_id=metadata_store_id,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )._gca_resource
+        gapic_artifact.metadata = metadata
+
+        resource_name = super().update(
+            resource_id=artifact_id,
+            resource_noun=cls._resource_noun,
+            gapic_resource=gapic_artifact,
+            metadata_store_id=metadata_store_id,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+        if not resource_name:
+            raise ValueError("Error while updating artifact")
+
+        return cls(
+            artifact_name=resource_name,
+            metadata_store_id=metadata_store_id,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+    @classmethod
+    def update_resource(
```

Comment:
Why do we need two update methods?

Reply:
I'm doing something similar than the create methods. As each resource needs to call its create_artifact, create_context, create_execution, I was basically having a base _Resource class that does pretty much everything, and the children have to implement the abstract create_resource (in this case update_resource) with their particular implementations (update_artifact, update_context, update_execution)

Might be a better way of doing this in python. Let me know your thoughts

Reply:
Remove update from child classes. Only needed in the parent _Resource. Did some other cleanups as well, to remove duplicated code, PTAL